meta-mender-raspberrypi/recipes-mender/update-firmware-state-script/files/ArtifactInstall_Leave_50

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+function trap_exit() {
+  umount --lazy /mnt/inactive_part
+}
+trap trap_exit EXIT
+
+echo "$(basename "$0") was called!" >&2
+
+# Check if mmcbkl0p2 is active part
+mount | grep /dev/mmcblk0p2
+if [ $? -eq 0 ]; then
+  inactive_part="/dev/mmcblk0p3"
+else
+  inactive_part="/dev/mmcblk0p2"
+fi
+
+mkdir -p /mnt/inactive_part
+mount -o ro ${inactive_part} /mnt/inactive_part
+
+# These are dangerous operations and if they fail (partial copy) it might
+# render the device unusable.
+
+# Copy 'core' firmware files first
+find /mnt/inactive_part/boot/firmware/ -maxdepth 1 -type f | xargs -I {} cp -v {} /uboot/
+
+# Synchronize before trying to copy the rest of the files
+sync
+
+# Copy overlays
+find /mnt/inactive_part/boot/firmware/overlays/ -maxdepth 1 -type f | xargs -I {} cp -v {} /uboot/overlays/
+
```

Comment:
I would consider adding a `sync` here too, to make sure the client does not leave the ArtifactInstall state without having everything in physical storage.

Reply:
Ack